homeassistant/components/flux_led/light.py

```diff
@@ -333,42 +415,62 @@ def turn_on(self, **kwargs):
         if brightness is None:
             brightness = self.brightness
 
-        # Preserve color on brightness/white level change
-        if rgb is None:
-            rgb = self._bulb.getRgb()
-
-        if white is None and self._mode == MODE_RGBW:
-            white = self.white_value
-
         # handle W only mode (use brightness instead of white value)
         if self._mode == MODE_WHITE:
             self._bulb.setRgbw(0, 0, 0, w=brightness)
+            return
+
+        if white is None and self._mode in WHITE_MODES:
+            white = self.white_value
+
+        # Preserve color on brightness/white level change
+        if rgb is None:
+            rgb = self._bulb.getRgb()
 
         # handle RGBW mode
-        elif self._mode == MODE_RGBW:
+        if self._mode in WHITE_MODES:
```

Comment:
Used here as well

Reply:
This won't work.  RGBWW requires setting the 5th channel which will require calculating the brightness. CCT, and RGB values.